src/k8s/pkg/k8sd/features/calico/status.go

```diff
@@ -6,42 +6,56 @@ import (
 
+	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// CheckNetwork checks the status of the Calico pods in the Kubernetes cluster.
-// It verifies if all the Calico pods in the "tigera-operator" namespace are ready.
-// If any pod is not ready, it returns false. Otherwise, it returns true.
-func CheckNetwork(ctx context.Context, snap snap.Snap) (bool, error) {
-	client, err := snap.KubernetesClient("calico-system")
-	if err != nil {
-		return false, fmt.Errorf("failed to create kubernetes client: %w", err)
+func podIsReady(pod v1.Pod) bool {
+	if pod.Status.Phase != v1.PodRunning {
+		return false
 	}
 
-	operatorReady, err := client.IsPodReady(ctx, "kube-system", "tigera-operator", metav1.ListOptions{})
-	if err != nil {
-		return false, fmt.Errorf("failed to get calico pods: %w", err)
-	}
-	if !operatorReady {
-		return false, nil
+	for _, condition := range pod.Status.Conditions {
+		if condition.Type == v1.PodReady && condition.Status == v1.ConditionTrue {
+			return true
+		}
 	}
 
-	calicoPods, err := client.ListPods(ctx, "calico-system", metav1.ListOptions{})
-	if err != nil {
-		return false, fmt.Errorf("failed to get calico pods: %w", err)
-	}
-	calicoApiserverPods, err := client.ListPods(ctx, "calico-apiserver", metav1.ListOptions{})
+	return false
+}
+
+// CheckNetwork checks the status of the Calico pods in the Kubernetes cluster.
+// We verify that the tigera-operator and calico-node pods are Ready and in Running state.
+func CheckNetwork(ctx context.Context, snap snap.Snap) (bool, error) {
+	client, err := snap.KubernetesClient("calico-system")
 	if err != nil {
-		return false, fmt.Errorf("failed to get calico-apiserver pods: %w", err)
+		return false, fmt.Errorf("failed to create kubernetes client: %w", err)
 	}
 
-	for _, pod := range append(calicoPods, calicoApiserverPods...) {
-		isReady, err := client.IsPodReady(ctx, pod.Name, "calico-system", metav1.ListOptions{})
+	for _, check := range []struct {
+		name      string
+		namespace string
+		labels    map[string]string
+	}{
+		// check that the tigera-operator pods are ready
+		{name: "tigera-operator", namespace: "tigera-operator", labels: map[string]string{"k8s-app": "tigera-operator"}},
+		// check that calico-node pods are ready
+		{name: "calico-node", namespace: "calico-system", labels: map[string]string{"app.kubernetes.io/name": "calico-node"}},
+	} {
+		pods, err := client.ListPods(ctx, check.namespace, metav1.ListOptions{
+			LabelSelector: metav1.FormatLabelSelector(&metav1.LabelSelector{MatchLabels: check.labels}),
+		})
 		if err != nil {
-			return false, fmt.Errorf("failed to check if pod %q is ready: %w", pod.Name, err)
+			return false, fmt.Errorf("failed to get %v pods: %w", check.name, err)
 		}
-		if !isReady {
-			return false, nil
+		if len(pods) == 0 {
+			return false, fmt.Errorf("no %v pods exist on the cluster", check.name)
+		}
+
+		for _, pod := range pods {
+			if !podIsReady(pod) {
```

Comment:
nit: instead of having this local function, we could refactor the `client.IsPodReady` or have a `client.PodsAreReady(ctx, namespace, ListOptions)` function. 
Feels a bit weird to have this generic pod checker in here.
Up to you

Reply:
yes, exactly, there is followup work to change how `client.ListPods` and `client.IsPodReady` to address the issues we discuss offline. Kept this out of the PR for brevity.